addons/binding/org.openhab.binding.mihome/src/main/java/org/openhab/binding/mihome/internal/api/manager/FailingRequestHandlerImpl.java

```diff
@@ -0,0 +1,89 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.mihome.internal.api.manager;
+
+import java.io.IOException;
+
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.openhab.binding.mihome.internal.api.JSONResponseHandler;
+import org.openhab.binding.mihome.internal.api.constants.JSONResponseConstants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.JsonObject;
+
+/**
+ * A {@link FailingRequestHandler} implementation that updates the status of a {@link Thing} using a
+ * {@link ThingCallback}
+ *
+ * @author Mihaela Memova
+ */
+
+public class FailingRequestHandlerImpl implements FailingRequestHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+    private ThingCallback callback;
+
+    public FailingRequestHandlerImpl(ThingCallback callback) {
+        this.callback = callback;
+    }
+
+    @Override
+    public void handleFailingHttpRequest(ContentResponse response) {
+        callback.updateThingStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, response.getReason());
+    }
+
+    @Override
+    public void handleFailingJsonRequest(JsonObject jsonResponse) {
+        String responseStatus = JSONResponseHandler.getResponseStatus(jsonResponse);
+        JsonObject responseData = jsonResponse.get(JSONResponseConstants.DATA_KEY).getAsJsonObject();
+        String errorMessage = JSONResponseHandler.getErrorMessageFromResponse(responseData);
+        switch (responseStatus) {
+            case JSONResponseConstants.RESPONSE_ACCESS_DENIED:
+                logger.error("Access to the requested action was not permitted");
+                callback.updateThingStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, errorMessage);
```

Comment:
In this case there is no duplication because the error message will be different depending on the case. Maybe logging the error is not necessary here.